.github/workflows/action-build-oci-tar-builder.yml

```diff
@@ -0,0 +1,41 @@
+name: Build oci-tar-builder
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        required: true
+        type: string
+
+jobs:
+  build:
+    name: build oci-tar-builder
+    runs-on: ${{ inputs.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        env:
+          RUST_CACHE_KEY_OS: rust-cache-${{ inputs.os }}
+        with:
+          rustflags: '' #Disable.  By default this action sets environment variable is set to -D warnings.  We manage this in the Makefile
+      - name: Build
+        run: make build-oci-tar-builder
+      - name: Run tests
+        timeout-minutes: 5
+        run: |
+          make test-oci-tar-builder
+      - name: Package artifacts
+        shell: bash
+        run: |
+          make dist-oci-tar-builder
+          # Check if there's any files to archive as tar fails otherwise
+          if stat dist/bin/* >/dev/null 2>&1; then
+            tar -czf dist/oci-tar-builder-${{ inputs.os }}.tar.gz -C dist/bin .
+          else
+            tar -czf dist/oci-tar-builder-${{ inputs.os }}.tar.gz -T /dev/null
+          fi
+      - name: Upload artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: oci-tar-builder-${{ inputs.os }}
```

Comment:
This would publish `oci-tar-builder-macos-12` and `oci-tar-builder-macos-14`
Could we have
* `macos-12` -> `macos-x86_64`
* `macos-14` -> `macos-aarch64`
* `ubuntu-22.04` -> `linux-glibc`
* `windows-latest` -> `windows`

or some similar mapping?

I think that would make finding the artifact to download wasier for our users.

Reply:
ah, wait, this is the CI artifact, not the published release, never mind then :-)